### PR TITLE
Image upload check

### PR DESCRIPTION
**Pull request by AshakaE**
# Check Image Upload

_This feature ensures file type being uploaded is an image with size less than 10mb_

## How should this be tested?

1. [postman](https://documenter.getpostman.com/view/13541794/UVsMvmHZ)
## Notes or observations

_The order for chaining the event handler on the readstream was wrong, that's why it was not throwing the error_

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/254

### CODE REVIEW

File: src/App/pinJSONAndImage/pin.resolver.ts [hunk 1]
```diff
@@ -19,20 +24,42 @@ class PinResolver {
...
+  private async checkFile(image: FileUpload, context: any) {
+    const type = image.mimetype.split('/')[0]
+    if (type !== 'image') {
+      Logger.error('Wrong file type')
+      throw new HttpException(
+        'Unsupported file type! Make sure file is an image.',
+        HttpStatus.BAD_REQUEST,
+      )
+    }
+    if (context.req.headers['content-length'] > 10000000) {
```
Comment (kesar) [inline src/App/pinJSONAndImage/pin.resolver.ts:36]: move the number on top of the class and define it as constant, that way its easier to read and easy to change.

validating the headers can be bypassed. So I would suggest add more robust methods like:

https://stackoverflow.com/questions/60518509/limit-file-size-when-upload-in-graphql & https://github.com/jaydenseric/graphql-upload/issues/204

Comment (kesar) [reply src/App/pinJSONAndImage/pin.resolver.ts:36]: why has you marked this as solved and re-request code review? perhaps you forgot to push to the branch? 🤔

Comment (AshakaE) [reply src/App/pinJSONAndImage/pin.resolver.ts:36]: Something on the tests not letting me push

File: src/App/pinJSONAndImage/pin.middleware.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { graphqlUploadExpress } from 'graphql-upload'
 @Injectable()
 class PinMiddleware implements NestMiddleware {
   use(req: any, res: any, next: () => void) {
-    graphqlUploadExpress({ maxFileSize: 5000000, maxFiles: 3 })(req, res, next)
+    graphqlUploadExpress({ maxFileSize: 10000000, maxFiles: 3 })(req, res, next)
```
Comment (kesar) [inline src/App/pinJSONAndImage/pin.middleware.ts:7]:
```suggestion
    graphqlUploadExpress({ maxFileSize: 2000000, maxFiles: 3 })(req, res, next)
```

in the specs i set 10MB but I dont see any reason to upload a file > 2MB 👍🏻

File: src/App/pinJSONAndImage/pin.resolver.ts [hunk 3]
```diff
@@ -19,23 +24,38 @@ class PinResolver {
...
+    return image
+  }
+
   @Mutation(() => IpfsHash, { name: 'pinImage' })
   async pinImage(
     @Args({ name: 'fileUpload', type: () => GraphQLUpload })
     image: FileUpload,
-  ): Promise<FileUpload> {
-    const { createReadStream, filename } = await image
+  ): Promise<any> {
```
Comment (kesar) [inline src/App/pinJSONAndImage/pin.resolver.ts:43]: is this not a FileUpload anymore? wondering why remove the type there

**Comment by kesar** [comment]
code looks fine, make sure its tested enough and pls tag @philipep-galdino for co-review / tests since you both are taking care of the backend. 

Justin C. will be focusing in the smart contracts, so its better if you both handle this 👍🏻

File: src/App/pinJSONAndImage/pin.resolver.ts (hunk 3 above)
Comment (AshakaE) [reply src/App/pinJSONAndImage/pin.resolver.ts]: This was an oversight 😶

File: src/App/pinJSONAndImage/pin.resolver.ts [hunk 4]
```diff
@@ -45,7 +45,8 @@ class PinResolver {
     const destinationPath = `./uploads/${filename}`
     return new Promise((res, rej) => {
       const read = createReadStream()
-      read.on('error', err => {
+      read.on('error', async err => {
+        await fs.unlink(destinationPath)
```
Comment (kesar) [inline src/App/pinJSONAndImage/pin.resolver.ts:49]: good catch

**Comment by AshakaE** [comment]
@philipep-galdino kindly review, I've tested the code, let me know of you find any anomalies 😆

**Review by philipep-galdino — APPROVED** [review]

**Pull request merged**